File: tradingagents/utils/logger.py
"""
美化的终端日志输出工具
"""
from datetime import datetime


class ColorLogger:
    """带颜色的日志输出"""
    
    # ANSI 颜色代码
    RESET = '\033[0m'
    BOLD = '\033[1m'
    
    # 前景色
    BLACK = '\033[30m'
    RED = '\033[31m'
    GREEN = '\033[32m'
    YELLOW = '\033[33m'
    BLUE = '\033[34m'
    MAGENTA = '\033[35m'
    CYAN = '\033[36m'
    WHITE = '\033[37m'
    
    # 高亮前景色
    BRIGHT_RED = '\033[91m'
    BRIGHT_GREEN = '\033[92m'
    BRIGHT_YELLOW = '\033[93m'
    BRIGHT_BLUE = '\033[94m'
    BRIGHT_MAGENTA = '\033[95m'
    BRIGHT_CYAN = '\033[96m'
    
    # 背景色
    BG_BLACK = '\033[40m'
    BG_RED = '\033[41m'
    BG_GREEN = '\033[42m'
    BG_YELLOW = '\033[43m'
    BG_BLUE = '\033[44m'
    BG_MAGENTA = '\033[45m'
    BG_CYAN = '\033[46m'
    BG_WHITE = '\033[47m'
    
    @classmethod
    def header(cls, text, char='=', width=80):
        """打印标题"""
        print(f"\n{cls.BOLD}{cls.BRIGHT_CYAN}{char * width}{cls.RESET}")
        print(f"{cls.BOLD}{cls.BRIGHT_CYAN}{text.center(width)}{cls.RESET}")
        print(f"{cls.BOLD}{cls.BRIGHT_CYAN}{char * width}{cls.RESET}\n")
    
    @classmethod
    def subheader(cls, text, char='─', width=80):
        """打印子标题"""
        print(f"\n{cls.BRIGHT_BLUE}{char * width}{cls.RESET}")
        print(f"{cls.BOLD}{cls.BRIGHT_BLUE}{text}{cls.RESET}")
        print(f"{cls.BRIGHT_BLUE}{char * width}{cls.RESET}\n")
    
    @classmethod
    def success(cls, text):
        """成功消息"""
        print(f"{cls.BRIGHT_GREEN}✅ {text}{cls.RESET}")
    
    @classmethod
    def error(cls, text):
        """错误消息"""
        print(f"{cls.BRIGHT_RED}❌ {text}{cls.RESET}")
    
    @classmethod
    def warning(cls, text):
        """警告消息"""
        print(f"{cls.BRIGHT_YELLOW}⚠️  {text}{cls.RESET}")
    
    @classmethod
    def info(cls, text):
        """信息消息"""
        print(f"{cls.CYAN}ℹ️  {text}{cls.RESET}")
    
    @classmethod
    def step(cls, step_num, text):
        """步骤消息"""
        print(f"{cls.BOLD}{cls.BRIGHT_MAGENTA}🔄 [步骤 {step_num}] {text}{cls.RESET}")
    
    @classmethod
    def tool_call(cls, tool_name):
        """工具调用"""
        print(f"{cls.YELLOW}🔧 调用工具: {cls.BOLD}{tool_name}{cls.RESET}")
    
    @classmethod
    def tool_result(cls, tool_name, result, max_lines=50):
        """工具结果"""
        print(f"\n{cls.BOLD}{cls.BG_BLUE}{cls.WHITE} Tool Message: {tool_name} {cls.RESET}")
        print(f"{cls.GREEN}{'─' * 80}{cls.RESET}")
        
        # 限制输出行数（如果太长）
        lines = result.split('\n')
        if len(lines) > max_lines:
            print('\n'.join(lines[:max_lines]))
            print(f"{cls.YELLOW}... (省略 {len(lines) - max_lines} 行){cls.RESET}")
        else:
            print(result)
        
        print(f"{cls.GREEN}{'─' * 80}{cls.RESET}\n")
    
    @classmethod
    def llm_response(cls, agent_name, content, max_lines=100):
        """LLM 响应"""
        print(f"\n{cls.BOLD}{cls.BG_MAGENTA}{cls.WHITE} {agent_name} LLM 响应 {cls.RESET}")
        print(f"{cls.MAGENTA}{'─' * 80}{cls.RESET}")
        
        lines = content.split('\n')
        if len(lines) > max_lines:
            print('\n'.join(lines[:max_lines]))
            print(f"{cls.YELLOW}... (省略 {len(lines) - max_lines} 行){cls.RESET}")
        else:
            print(content)
        
        print(f"{cls.MAGENTA}{'─' * 80}{cls.RESET}\n")
    
    @classmethod
    def position_info(cls, info):
        """持仓信息"""
        print(f"\n{cls.BOLD}{cls.BG_CYAN}{cls.BLACK} 💼 账户和持仓信息 {cls.RESET}")
        print(f"{cls.CYAN}{'─' * 80}{cls.RESET}")
        print(info)
        print(f"{cls.CYAN}{'─' * 80}{cls.RESET}\n")
    
    @classmethod
    def decision(cls, decision_text):
        """最终决策"""
        print(f"\n{cls.BOLD}{cls.BG_GREEN}{cls.BLACK} ✅ 最终交易决策 {cls.RESET}")
        print(f"{cls.GREEN}{'=' * 80}{cls.RESET}")
        print(decision_text)
        print(f"{cls.GREEN}{'=' * 80}{cls.RESET}\n")
    
    @classmethod
    def timestamp(cls):
        """时间戳"""
        return f"{cls.CYAN}[{datetime.now().strftime('%H:%M:%S')}]{cls.RESET}"

File: internal/logger/logger.go
package logger

import (
	"fmt"
	"io"
	"os"
	"strings"
	"time"

	"github.com/rs/zerolog"
)

// ANSI color codes
const (
	Reset  = "\033[0m"
	Bold   = "\033[1m"

	// Foreground colors
	Red     = "\033[31m"
	Green   = "\033[32m"
	Yellow  = "\033[33m"
	Blue    = "\033[34m"
	Magenta = "\033[35m"
	Cyan    = "\033[36m"
	White   = "\033[37m"

	// Bright foreground colors
	BrightRed     = "\033[91m"
	BrightGreen   = "\033[92m"
	BrightYellow  = "\033[93m"
	BrightBlue    = "\033[94m"
	BrightMagenta = "\033[95m"
	BrightCyan    = "\033[96m"

	// Background colors
	BgBlack   = "\033[40m"
	BgRed     = "\033[41m"
	BgGreen   = "\033[42m"
	BgYellow  = "\033[43m"
	BgBlue    = "\033[44m"
	BgMagenta = "\033[45m"
	BgCyan    = "\033[46m"
	BgWhite   = "\033[47m"
)

// ColorLogger provides colored terminal output
type ColorLogger struct {
	logger zerolog.Logger
	writer io.Writer
}

// NewColorLogger creates a new ColorLogger instance
func NewColorLogger(debug bool) *ColorLogger {
	output := zerolog.ConsoleWriter{
		Out:        os.Stdout,
		TimeFormat: time.RFC3339,
		NoColor:    false,
	}

	zerolog.SetGlobalLevel(zerolog.InfoLevel)
	if debug {
		zerolog.SetGlobalLevel(zerolog.DebugLevel)
	}

	logger := zerolog.New(output).With().Timestamp().Logger()

	return &ColorLogger{
		logger: logger,
		writer: os.Stdout,
	}
}

// Header prints a header with the given text
func (l *ColorLogger) Header(text string, char rune, width int) {
	line := strings.Repeat(string(char), width)
	fmt.Fprintf(l.writer, "\n%s%s%s%s\n", Bold, BrightCyan, line, Reset)
	fmt.Fprintf(l.writer, "%s%s%s%s\n", Bold, BrightCyan, center(text, width), Reset)
	fmt.Fprintf(l.writer, "%s%s%s%s\n\n", Bold, BrightCyan, line, Reset)
}

// Subheader prints a subheader
func (l *ColorLogger) Subheader(text string, char rune, width int) {
	line := strings.Repeat(string(char), width)
	fmt.Fprintf(l.writer, "\n%s%s%s\n", BrightBlue, line, Reset)
	fmt.Fprintf(l.writer, "%s%s%s%s\n", Bold, BrightBlue, text, Reset)
	fmt.Fprintf(l.writer, "%s%s%s\n\n", BrightBlue, line, Reset)
}

// Success prints a success message
func (l *ColorLogger) Success(text string) {
	fmt.Fprintf(l.writer, "%s✅ %s%s\n", BrightGreen, text, Reset)
	l.logger.Info().Msg(text)
}

// Error prints an error message
func (l *ColorLogger) Error(text string) {
	fmt.Fprintf(l.writer, "%s❌ %s%s\n", BrightRed, text, Reset)
	l.logger.Error().Msg(text)
}

// Warning prints a warning message
func (l *ColorLogger) Warning(text string) {
	fmt.Fprintf(l.writer, "%s⚠️  %s%s\n", BrightYellow, text, Reset)
	l.logger.Warn().Msg(text)
}

// Info prints an info message
func (l *ColorLogger) Info(text string) {
	fmt.Fprintf(l.writer, "%sℹ️  %s%s\n", Cyan, text, Reset)
	l.logger.Info().Msg(text)
}

// Step prints a step message
func (l *ColorLogger) Step(stepNum int, text string) {
	fmt.Fprintf(l.writer, "%s%s🔄 [步骤 %d] %s%s\n", Bold, BrightMagenta, stepNum, text, Reset)
	l.logger.Info().Int("step", stepNum).Msg(text)
}

// ToolCall prints a tool call message
func (l *ColorLogger) ToolCall(toolName string) {
	fmt.Fprintf(l.writer, "%s🔧 调用工具: %s%s%s\n", Yellow, Bold, toolName, Reset)
	l.logger.Debug().Str("tool", toolName).Msg("Tool called")
}

// ToolResult prints a tool result
func (l *ColorLogger) ToolResult(toolName string, result string, maxLines int) {
	fmt.Fprintf(l.writer, "\n%s%s%s Tool Message: %s %s\n", Bold, BgBlue, White, toolName, Reset)
	fmt.Fprintf(l.writer, "%s%s%s\n", Green, strings.Repeat("─", 80), Reset)

	lines := strings.Split(result, "\n")
	if len(lines) > maxLines {
		fmt.Fprintln(l.writer, strings.Join(lines[:maxLines], "\n"))
		fmt.Fprintf(l.writer, "%s... (省略 %d 行)%s\n", Yellow, len(lines)-maxLines, Reset)
	} else {
		fmt.Fprintln(l.writer, result)
	}

	fmt.Fprintf(l.writer, "%s%s%s\n\n", Green, strings.Repeat("─", 80), Reset)
}

// LLMResponse prints an LLM response
func (l *ColorLogger) LLMResponse(agentName string, content string, maxLines int) {
	fmt.Fprintf(l.writer, "\n%s%s%s %s LLM 响应 %s\n", Bold, BgMagenta, White, agentName, Reset)
	fmt.Fprintf(l.writer, "%s%s%s\n", Magenta, strings.Repeat("─", 80), Reset)

	lines := strings.Split(content, "\n")
	if len(lines) > maxLines {
		fmt.Fprintln(l.writer, strings.Join(lines[:maxLines], "\n"))
		fmt.Fprintf(l.writer, "%s... (省略 %d 行)%s\n", Yellow, len(lines)-maxLines, Reset)
	} else {
		fmt.Fprintln(l.writer, content)
	}

	fmt.Fprintf(l.writer, "%s%s%s\n\n", Magenta, strings.Repeat("─", 80), Reset)
}

// PositionInfo prints position information
func (l *ColorLogger) PositionInfo(info string) {
	fmt.Fprintf(l.writer, "\n%s%s%s 💼 账户和持仓信息 %s\n", Bold, BgCyan, White, Reset)
	fmt.Fprintf(l.writer, "%s%s%s\n", Cyan, strings.Repeat("─", 80), Reset)
	fmt.Fprintln(l.writer, info)
	fmt.Fprintf(l.writer, "%s%s%s\n\n", Cyan, strings.Repeat("─", 80), Reset)
}

// Decision prints the final trading decision
func (l *ColorLogger) Decision(decisionText string) {
	fmt.Fprintf(l.writer, "\n%s%s%s ✅ 最终交易决策 %s\n", Bold, BgGreen, White, Reset)
	fmt.Fprintf(l.writer, "%s%s%s\n", Green, strings.Repeat("=", 80), Reset)
	fmt.Fprintln(l.writer, decisionText)
	fmt.Fprintf(l.writer, "%s%s%s\n\n", Green, strings.Repeat("=", 80), Reset)
}

// Timestamp returns a formatted timestamp
func (l *ColorLogger) Timestamp() string {
	return fmt.Sprintf("%s[%s]%s", Cyan, time.Now().Format("15:04:05"), Reset)
}

// Debug prints a debug message (only if debug mode is enabled)
func (l *ColorLogger) Debug(text string) {
	l.logger.Debug().Msg(text)
}

// Helper function to center text
func center(text string, width int) string {
	if len(text) >= width {
		return text
	}
	padding := (width - len(text)) / 2
	return strings.Repeat(" ", padding) + text
}

// Global logger instance
var Global *ColorLogger

// Init initializes the global logger
func Init(debug bool) {
	Global = NewColorLogger(debug)
}